xonsh/environ.py

```diff
@@ -691,18 +691,36 @@ def locate_binary(name):
         return None
 
 
+def _get_parent_dir_for(cwd, dir_name):
+    # walk up the directory tree to see if we are inside an hg repo
+    dir_list = [dir_name]
+    path = cwd.split(os.path.sep)
```

Comment:
I don't know why most of this looks like it does. I don't think that it was me:)